Userland/Applications/NetworkSettings/NetworkSettingsWidget.cpp

```diff
@@ -134,33 +137,73 @@ void NetworkSettingsWidget::on_switch_enabled_or_dhcp()
 
 void NetworkSettingsWidget::apply_settings()
 {
-    auto config_file = Core::ConfigFile::open_for_system("Network", Core::ConfigFile::AllowWriting::Yes).release_value_but_fixme_should_propagate_errors();
+    auto result = apply_settings_impl();
+    if (result.is_error()) {
+        GUI::MessageBox::show_error(window(), result.release_error().string_literal());
+        return;
+    }
+}
+
+ErrorOr<void> NetworkSettingsWidget::apply_settings_impl()
+{
+    auto maybe_json = TRY(create_settings_object());
+    if (!maybe_json.has_value() || maybe_json.value().is_empty())
+        return {};
+    auto json = maybe_json.release_value();
+
+    auto pipefds = TRY(Core::System::pipe2(O_CLOEXEC));
+    ScopeGuard guard_fd1 { [&] { close(pipefds[1]); } };
+    {
+        posix_spawn_file_actions_t file_actions;
+        posix_spawn_file_actions_init(&file_actions);
+        posix_spawn_file_actions_adddup2(&file_actions, pipefds[0], STDIN_FILENO);
+
+        ScopeGuard guard_fd0_and_file_actions { [&]() {
+            posix_spawn_file_actions_destroy(&file_actions);
+            close(pipefds[0]);
+        } };
+
+        char const* argv[] = { "/bin/Escalator", "-I", "-P", "To apply these changes please enter your users password:", "/bin/network-settings", nullptr };
```

Comment:
"your users password" sounds wrong, please change to either "your userpassword" or just "your password".